doc/west/repo-tool.rst

```diff
@@ -0,0 +1,266 @@
+
+  With ``-b``, creates and checks out the branch if it does not exist in a
+  project (like ``git checkout -b``), instead of ignoring the project.
+
+Miscellanous commands
+*********************
+
+These commands perform miscellaneous functions.
+
+- ``west list``: Lists project information from the manifest (URL, revision,
+  path, etc.), along with other manifest-related information.
+
+- ``west diff [PROJECT ...] [ARGUMENT ...]``: Runs a multi-repo ``git diff``
+  for the specified projects (default: all projects).
+
+  Extra arguments are passed as-is to ``git diff``.
+
+- ``west status [PROJECT ...] [ARGUMENT ...]``: Like ``west diff``, for
+  running ``git status``.
+
+- ``west update [--update-west] [--update-manifest]``: Updates the west and
+  manifest repositories.
+
+  Normally, the west and manifest repositories are updated automatically
+  whenever a command that fetches upstream data is run. This self-update is
+  skipped (with a warning) if not possible via fast-forward (i.e., if you've
+  made modifications to the west or manifest repositories).
+
+Example workflow
+================
+
+This section gives an example workflow that clones the Zephyr repositories,
+creates and switches between two work branches, updates (rebases) a work branch
+to the latest manifest revision, and inspects upstream changes.
+
+1. First, we initialize West with the default Zephyr manifest, and fetch all
+   projects:
+
+   .. code-block:: console
+
+      $ west init
+      $ west fetch
+
+   .. note::
+
+      ``west clone`` could be used instead of ``west fetch``, to automatically
+      create and check out a branch in each project, named after the project's
+      revision. In this example, we create all branches ourselves.
+
+2. Next, we create and check out a branch ``xy-feature`` for implementing a
+   feature that spans the ``proj-x`` and ``proj-y`` projects:
+
+   .. code-block:: console
+
+      $ west checkout -b xy-feature proj-x proj-y
+
+   This creates a Git branch named ``xy-feature`` in ``proj-x`` and ``proj-y``.
+   The new branches are automatically set up to track the ``manifest-rev``
+   branch in their respective repositories.
+
+   We work on the repositories as usual.
+
+3. While working on the feature, an urgent bug comes up that spans all
+   projects. We commit/stash the partially implemented feature and check out a
+   new branch for fixing it:
+
+   .. code-block:: console
+
+      $ west checkout -b global-fix
+
+4. After fixing the bug, we want to switch back to working on the feature, but
+   we forgot its branch name. We list branches with ``west branch`` to remind
+   ourselves, and then switch back:
+
+   .. code-block:: console
+
+      $ west checkout xy-feature
+
+   This command only affects the ``proj-x`` and ``proj-y`` repositories, since
+   only those have the ``xy-feature`` branch.
+
+5. We now update (rebase) the ``proj-1`` and ``proj-2`` repositories, to get
```

Comment:
Looks like there's some leftovers from a previous version that used `proj-NUMBER` instead of `proj-LETTER` (elsewhere in the file too).

Reply:
Fixed